tests/conftest.py

```diff
@@ -54,3 +58,16 @@ def pytest_addoption(parser):
         '--legacy_password',
         default=os.getenv('LEGACY_PASSWORD'),
         help='password for CNX legacy.')
+
+
+# https://docs.pytest.org/en/latest/example/simple.html#making-test-result-information-available-in-fixtures
+@pytest.hookimpl(tryfirst=True, hookwrapper=True)
+def pytest_runtest_makereport(item, call):
+    # execute all other hooks to obtain the report object
+    outcome = yield
+    rep = outcome.get_result()
+
+    # set a report attribute for each phase of a call, which can
+    # be "setup", "call", "teardown"
+
+    setattr(item, "rep_" + rep.when, rep)
```

Comment:
string concatenation is generally frowned upon in python. You'll want to use `'rep_{}'.format(rep.when)`.